src/server/nyplApiClient/index.ts

```diff
@@ -46,28 +22,33 @@ const nyplApiClient = async (options = { apiName: "platform" }) => {
   }
 
   const baseUrl = appConfig.apiEndpoints[apiName][appEnvironment]
-  return await new Promise((resolve, reject) => {
-    Promise.all(keys.map(decryptKMS))
-      .then(([decryptedClientId, decryptedClientSecret]) => {
-        const nyplApiClient = new NyplApiClient({
-          base_url: baseUrl,
-          oauth_key: decryptedClientId,
-          oauth_secret: decryptedClientSecret,
-          oauth_url: appConfig.urls.tokenUrl,
-        })
-
-        CACHE.clientId = clientId
-        CACHE.clientSecret = clientSecret
-        CACHE.clients[apiName] = nyplApiClient
 
-        resolve(nyplApiClient)
-      })
-      .catch((error) => {
-        // TODO: add this once the Winston config is added.
-        // logger.error('ERROR trying to decrypt using KMS.', error);
-        reject(error)
-      })
-  })
+  let decryptedId: string
+  let decryptedSecret: string
+  if (CACHE.secret && CACHE.id) {
+    decryptedId = CACHE.id
+    decryptedSecret = CACHE.secret
+  } else {
+    try {
+      ;[decryptedId, decryptedSecret] = await kmsDecryptCreds(creds)
+      CACHE.id = decryptedId
+      CACHE.secret = decryptedSecret
+    } catch (exception) {
+      console.error("Error decrypting Sierra credentials")
+    }
+  }
+  try {
+    const nyplApiClient = new NyplApiClient({
+      base_url: baseUrl,
+      oauth_key: decryptedId,
+      oauth_secret: decryptedSecret,
+      oauth_url: appConfig.urls.tokenUrl,
+    })
+    CACHE.clients[apiName] = nyplApiClient
+    return nyplApiClient
+  } catch (error) {
+    console.error(error.message)
```

Comment:
Not necessarily needed for this PR but this should throw an error. If something goes wrong here, calling `client.get("...")` will crash